### ISSUE TITLE
Tribler should be able to recover in case of corrupted files/databases

### ISSUE THREAD

**Issue by whirm** [Tribler/tribler#1993]
To save our users situations like #1992

**Comment by whirm** [comment]
Another example:

```
ERROR   1460769681.46        dispersy:1724  exception during handle_callback for torrent-collect-request
Traceback (most recent call last):
  File "Tribler\dispersy\dispersy.pyo", line 1719, in _update
  File "Tribler\community\search\community.pyo", line 466, in on_torrent_collect_request
  File "Tribler\community\search\community.pyo", line 512, in _create_pingpong
  File "Tribler\community\search\community.pyo", line 537, in __get_torrents
  File "Tribler\Core\CacheDB\SqliteCacheDBHandler.pyo", line 1370, in getRecentlyCollectedSwiftHashes
  File "Tribler\dispersy\util.pyo", line 43, in helper
  File "Tribler\Core\CacheDB\sqlitecachedb.pyo", line 2296, in fetchall
  File "Tribler\Core\CacheDB\sqlitecachedb.pyo", line 574, in fetchall
  File "Tribler\Core\CacheDB\sqlitecachedb.pyo", line 485, in execute_read
  File "Tribler\dispersy\util.pyo", line 43, in helper
  File "Tribler\Core\CacheDB\sqlitecachedb.pyo", line 2317, in _execute
CorruptError: CorruptError: database disk image is malformed
ERROR   1460769684.46   sqlitecachedb:2316  ===Twisted===
SQL Type: <type 'str'>
-----

            SELECT CT.swift_torrent_hash, CT.infohash, CT.num_seeders, CT.num_leechers, T.last_tracker_check, CT.insert_time
             FROM Torrent T, CollectedTorrent CT
             WHERE CT.torrent_id = T.torrent_id
             AND CT.swift_torrent_hash IS NOT NULL AND CT.swift_torrent_hash <> ''
             AND T.secret is not 1 ORDER BY CT.insert_time DESC LIMIT ?
             
-----
```

**Comment by kozlovsky** [comment]
Fixed in #7628
